BLEN-216: Matlib implementation

VascoPi opened the pull request:
Added mathlib module, with ui.py and manager.py.
Added properties module.
Added class MaterialXProperties to be used as a parent for addon PropertyGroups. It sets "materialx" property as parent for addon properties (bpy.context.window_manager.materialx. ...) Added constant variables MATLIB_FOLDER, MATLIB_DIR, currently, it is set to save matlib to addon appdata folder.

VascoPi committed:
BLEN-216: Matlib implementation.
Added mathlib module, with ui.py and manager.py.
Added properties module.
Added class MaterialXProperties to be used as parent for addon PropertyGroups.
It sets "materialx" proprty as parent for addon properties (bpy.context.window_manager.materialx. ...)
Added constant variables MATLIB_FOLDER, MATLIB_DIR, currently, it is set to save matlib to addon appdata folder.

bnagirniak commented on materialx/properties/__init__.py:
```diff
@@ -0,0 +1,34 @@
+import bpy
+
+
+class MaterialXProperties(bpy.types.PropertyGroup):
```
Move to utils.py

bnagirniak commented on materialx/properties/matlib.py:
```diff
@@ -0,0 +1,145 @@
+log = logging.Log(tag='properties.matlib')
+
+
+class MatlibProperties(bpy.types.PropertyGroup):
```
move to matlib/properties.py

VascoPi committed:
Moved properties to matlib folder.
Removed module properties.
Moved MaterialXProperties to utils.py.
Log fixes.
Code cleanup.

bnagirniak committed:
Merge branch 'materialx-addon' into BLEN-216

bnagirniak committed:
Small improvements

bnagirniak reviewed (approved):
Works good.
